http3/response_writer.go

```diff
@@ -50,6 +51,41 @@ func (w *responseWriter) Header() http.Header {
 	return w.header
 }
 
+// flushHeader send header frame in wire format when:
+// 1. 1XX status is written
+// 1. Flush is called
+// 2. Write can not longer buffer anymore data
+func (w *responseWriter) flushHeader() error {
+	var headers bytes.Buffer
+	// leave some room to encode frame header
+	headers.Write(w.buf[:cap(w.buf)])
```

Comment:
Is there a better way? This doesn't seem clean.

Reply:
This part is very hacky, I'll admit. But I also used a similar technique when refactoring caddy's [fastcgi transport](https://github.com/caddyserver/caddy/pull/4978/files#diff-384ea25b96b47d84c242c7fe3341847cb468d867fafdea97d0920b605321a35dR120). Although fastcgi's length field is fixed size (2 bytes), and quic uses a variable length with a maximum size of 16 bytes.

The alternative is to use another buffer, but it's unnecessary. Since there may be data already buffered in `bufferedStr`, we can use that either.

I tend to not use too many hacks like this, this is one of the place I think it is sensible to use.